subsys/bluetooth/host/adv.c

```diff
@@ -978,6 +992,11 @@ int bt_le_adv_start_ext(struct bt_le_ext_adv *adv,
 	return 0;
 }
 
+static void adv_timeout(struct k_work *work)
+{
+	bt_le_adv_stop();
+}
```

Comment:
This will only work for a single advertising set. With extended advertising we can have multiple sets, and we need to find and stop the right one.

Reply:
I updated a PR with your suggestions, if I understand correctly, I can get the right set by using `CONTAINER_OF` macro in `work` function?